web/src/features/Gitops/components/GitOpsRepoDetails.jsx

```diff
@@ -0,0 +1,212 @@
+        <Flex flex="1" mt="30" mb="20" width="100%">
+          {provider !== "other" && (
+            <div className="flex flex1 flex-column u-marginRight--20">
+              <p className="u-fontSize--large u-textColor--primary u-fontWeight--bold u-lineHeight--normal">
+                {isBitbucketServer ? "Project" : "Owner"}
+                <span> (Required)</span>
+              </p>
+              <input
+                type="text"
+                className={`Input ${
+                  providerError?.field === "owner" && "has-error"
+                }`}
+                placeholder={isBitbucketServer ? "project" : "owner"}
+                value={owner}
+                onChange={(e) => setOwner(e.target.value)}
+                autoFocus
+              />
+              {providerError?.field === "owner" && (
+                <p className="u-fontSize--small u-marginTop--5 u-color--chestnut u-fontWeight--medium u-lineHeight--normal">
+                  {isBitbucketServer
+                    ? "A project must be provided"
+                    : "An owner must be provided"}
+                </p>
+              )}
+            </div>
+          )}
+
+          {provider !== "other" && (
+            <Flex flex="1" direction="column">
+              <Paragraph
+                size="16"
+                weight="bold"
+                className="u-lineHeight--normal"
+              >
+                Repository <span>(Required)</span>
+              </Paragraph>
+              <input
+                type="text"
+                className={`Input ${
+                  providerError?.field === "repo" && "has-error"
+                }`}
+                placeholder={"Repository"}
+                value={repo}
+                onChange={(e) => setRepo(e.target.value)}
+                autoFocus
+              />
+              {providerError?.field === "owner" && (
+                <p className="u-fontSize--small u-marginTop--5 u-color--chestnut u-fontWeight--medium u-lineHeight--normal">
+                  A repository must be provided
+                </p>
+              )}
+            </Flex>
+          )}
+        </Flex>
+        <Flex width="100%">
+          {provider !== "other" && (
+            <div className="flex flex1 flex-column u-marginRight--20">
+              <p className="u-fontSize--large u-textColor--primary u-fontWeight--bold u-lineHeight--normal">
+                Branch
+              </p>
+              <p className="u-fontSize--normal u-textColor--bodyCopy u-fontWeight--medium u-lineHeight--normal u-marginBottom--10">
+                Leave blank to use the default branch.
+              </p>
+              <input
+                type="text"
+                className={`Input`}
+                placeholder="main"
+                value={branch}
+                onChange={(e) => setBranch(e.target.value)}
+              />
+            </div>
+          )}
+          {provider !== "other" && (
```

Comment:
Is there any particular reason we are keeping these blocks inside separate `provider !== "other"` checks? Could they be condensed into one?

Reply:
thanks for catching that. I believe the check is unnecessary looking back. Provider would never equal to other as it isn't even a choice.